compiler/src/dotty/tools/dotc/transform/TypeUtils.scala

```diff
@@ -58,10 +58,15 @@ object TypeUtils {
         if (arity < 0) arity else arity + 1
       case self: SingletonType =>
         if self.termSymbol == defn.EmptyTupleModule then 0 else -1
-      case self if defn.isTupleClass(self.classSymbol) =>
-        self.dealias.argInfos.length
+      case AndType(tp1, tp2) =>
+        tp1.tupleArity.max(tp2.tupleArity)
```

Comment:
should this check that both sides are equal? `(Int, Int) & (Int, Int, Int)` doesn't make sense to me